Extension/package.json

```diff
@@ -6121,24 +6121,25 @@
         "scripts": "ts-node -T .scripts/scripts.ts",
         "show": "ts-node -T .scripts/clean.ts show",
         "clean": "ts-node -T .scripts/clean.ts",
-        "test": "ts-node -T .scripts/test.ts",
-        "code": "ts-node -T .scripts/code.ts",
+        "test": "yarn install && (yarn verify prep --quiet || yarn prep) && (yarn verify compiled --quiet || yarn build) && ts-node -T .scripts/test.ts",
+        "code": "yarn install && (yarn verify compiled --quiet || yarn build) && yarn verify binaries && ts-node -T .scripts/code.ts",
         "verify": "ts-node -T .scripts/verify.ts",
-        "prep": "yarn copy-walkthrough-media && yarn generate-native-strings && yarn translations-generate",
-        "lint": "eslint -c .eslintrc.js --report-unused-disable-directives src test ui .scripts",
-        "compile": "(yarn verify prep || yarn prep) && tsc --build tsconfig.json",
-        "watch": "(yarn verify prep || yarn prep )&& tsc --build tsconfig.json --watch",
-        "rebuild": "yarn clean && yarn prep && yarn compile",
+        "prep": "yarn prep:dts && yarn copy-walkthrough-media && yarn generate-native-strings && yarn translations-generate",
+        "lint": "yarn install && eslint -c .eslintrc.js --report-unused-disable-directives src test ui .scripts",
+        "compile": "yarn install && (yarn verify prep --quiet || yarn prep) && yarn build",
+        "watch": "yarn install && (yarn verify prep --quiet || yarn prep) && tsc --build tsconfig.json --watch",
+        "rebuild": "yarn install && yarn clean && yarn prep && yarn build",
         "vscode:prepublish": "yarn clean && yarn webpack",
```

Comment:
Do you want to move the `translations-generate` step to `vscode:prepublish` only?  Or may a separate version of `compile`/`webpack` that someone could run only if they really need localized strings generated (and which `vscode:prepublish` would leverage, so happens for the VSIX).  It looks like you're now checking to ensure this only happens once now?  FYI, locally, it may be needed only zero times.  :)

Reply:
It's only doing it once and it's pretty good at making sure that it's not being done needlessly.
I didn't want to explode to more tasks, or complicate things further at this point. 

I suspect we may have more opportunities to refine this as I go look at the other pipelines anyway.